src/ethernet_interface.cpp

```diff
@@ -319,6 +319,20 @@ ObjectPath EthernetInterface::ip(IP::Protocol protType, std::string ipaddress,
 
     std::string objectPath =
         generateObjectPath(protType, ipaddress, prefixLength, gateway);
+
+    auto it = this->addrs.find(ipaddress);
+    if (it != this->addrs.end())
+    {
+        if (it->second->getObjPath() == objectPath)
+        {
+            return it->second->getObjPath();
```

Comment:
Are you able to modify the gateway ip for an existing IP address set ? Please test that and update the `Tested by` section.

Reply:
> Are you able to modify the gateway ip for an existing IP address set ? Please test that and update the `Tested by` section.

Yes, I have already tested it but missed to add in the description under Tested by section.

Reply:
@nkotania ipaddress is a string which you are using in the find right?

Reply:
> @nkotania ipaddress is a string which you are using in the find right?

Yes